Debugger Debbie follow-up after going deeper on the cross-repo boundary.

My earlier review framing was too broad: **the `_EVENT_TYPE_TO_MODEL` addition is not what starts SaaS queueing these events.** That hypothesis is falsified.

## Falsified hypothesis: model-map registration causes queueing

I installed the sibling `spec-kitty-events` checkout into the `spec-kitty` PR #1926 venv and compared behavior with events `6.1.0` vs PyPI `6.0.0`.

With this PR installed (`6.1.0`):

```text
events_version 6.1.0
model_map_has_reopened True
model_map_has_followup True
queue_event_is_none False
queue_event_type MissionReopened
queue_payload_keys ['cleared_merge', 'mission_id', 'mission_slug', 'reason', 'reopened_at', 'reopened_by']
```

Then with `spec-kitty-events==6.0.0` restored:

```text
events_version 6.0.0
model_map_has_reopened False
model_map_has_followup False
queue_event_is_none False
queue_event_type MissionReopened
```

So the producer already builds a SaaS queue event for `MissionReopened` even when the external events package does not recognize the type. The model-map registration changes validation/contract recognition, but not the queue/no-queue decision.

I opened the SaaS follow-up requested by Robert for projection support: https://github.com/Priivacy-ai/spec-kitty-saas/issues/318

## Actual blocker: reducer semantics now classify valid post-mission events as anomalies

This PR adds `MissionReopened` and `FollowUpRecorded` to `MISSION_EVENT_TYPES` in `src/spec_kitty_events/lifecycle.py`, which means `reduce_lifecycle_events()` now includes them in `mission_events`:

```python
mission_events = [
    e for e in unique_events if e.event_type in MISSION_EVENT_TYPES
]
```

But `_process_mission_event()` still has the generic terminal-state guard before any handler for these two new post-mission event types:

```python
# Check: event after terminal state
if mission_status in TERMINAL_MISSION_STATUSES:
    anomalies.append(
        LifecycleAnomaly(
            event_id=event.event_id,
            event_type=event.event_type,
            reason=f"Event after terminal state ({mission_status})",
        )
    )
    return mission_id, mission_status, mission_type, current_phase
```

Because these events are explicitly post-mission lifecycle facts, `MissionStarted -> MissionCompleted -> MissionReopened` and `MissionStarted -> MissionCompleted -> FollowUpRecorded` should not be anomalies.

## Repro proof

I ran a direct reducer smoke against this PR.

For `MissionReopened` after completion:

```text
mission_status MissionStatus.COMPLETED
event_count 3
last_processed_event_id 33333333-3333-3333-3333-333333333333
anomalies [('33333333-3333-3333-3333-333333333333', 'MissionReopened', 'Event after terminal state (MissionStatus.COMPLETED)')]
```

For `FollowUpRecorded` after completion:

```text
mission_status MissionStatus.COMPLETED
anomalies [('33333333-3333-3333-3333-333333333333', 'FollowUpRecorded', 'Event after terminal state (MissionStatus.COMPLETED)')]
```

That means the canonical events package accepts these event payloads, but its own lifecycle reducer reports normal intended usage as anomalous.

## Cross-repo evidence

`spec-kitty` PR #1926 currently has a boundary test that fails with this PR installed:

```text
FAILED tests/specify_cli/status/test_post_mission_lifecycle_events.py::test_new_types_not_on_saas_strict_path
E   AssertionError: assert 'MissionReopened' not in {...}
```

That specific assertion is based on the old “local-only this mission” assumption. If #41 intentionally canonicalizes these events, that test should change later in the CLI consuming PR. It is not the core bug by itself.

The core bug is reducer semantics: once these are in `MISSION_EVENT_TYPES`, the reducer must understand them as legitimate post-terminal facts.

## Suggested fix before PyPI `6.1.0`

1. Add explicit `_process_mission_event()` handling for `MissionReopenedPayload` and `FollowUpRecordedPayload`.
2. Handle those before the terminal-state guard, or make the terminal guard exclude these two post-mission event types.
3. `MissionReopened`: validate payload and decide the canonical reducer state. Given the event name, likely set `mission_status = MissionStatus.ACTIVE`; at minimum, do not produce an anomaly.
4. `FollowUpRecorded`: validate payload, no status mutation, no anomaly.
5. Add reducer tests for:
   - completed -> reopened is accepted
   - completed -> follow-up is accepted
   - invalid reopened/follow-up payloads produce the appropriate invalid-payload anomaly
   - cancel-vs-reopen precedence, since the reducer docstring already says “cancel-beats-re-open precedence”

Local checks on this PR still pass as-is:

```text
pytest -q: 2116 passed
mypy src/: success
```

Those checks do not currently cover the post-terminal reducer path for the new events, which is the missing gate.

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -27,6 +27,20 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
   model-level validator enforces the commit-vs-pr conditional-required rule.
   Consumer: spec-kitty mission `01KV0S99` (PR Priivacy-ai/spec-kitty#1926).
   No JSON-schema entry yet (the schema layer is optional secondary).
+- **`reduce_lifecycle_events` post-mission semantics** for the two new events.
+  Because `MissionReopened`/`FollowUpRecorded` are members of
+  `MISSION_EVENT_TYPES`, they now flow through the lifecycle reducer with
+  explicit handlers placed *before* the generic post-terminal guard (which would
+  otherwise misfire and flag them as `Event after terminal state` anomalies).
+  A `MissionReopened` is valid only when the mission is terminal and transitions
+  it to the new actionable `MissionStatus.REOPENED` state (non-terminal, so a
+  fresh `MissionCompleted` is processed normally); a `FollowUpRecorded` is valid
+  only when terminal and leaves `mission_status` unchanged (a recorded fact).
+  Inverse contract: either event arriving before completion (mission not in a
+  terminal state) is itself flagged as a `… before completion` anomaly. All
+  other event semantics and existing anomaly detection are preserved.
+- `MissionStatus.REOPENED = "reopened"` enum member (actionable, NOT in
+  `TERMINAL_MISSION_STATUSES`).
 
 ## [6.0.0] - 2026-06-07
 

--- a/src/spec_kitty_events/lifecycle.py
+++ b/src/spec_kitty_events/lifecycle.py
@@ -59,11 +59,20 @@ MISSION_EVENT_TYPES: FrozenSet[str] = frozenset({
 
 
 class MissionStatus(str, Enum):
-    """Mission lifecycle states."""
+    """Mission lifecycle states.
+
+    ``REOPENED`` is an *actionable* (non-terminal) state distinct from
+    ``ACTIVE``: it records that a previously-completed/cancelled mission was
+    returned to an actionable state by a ``MissionReopened`` event. It is
+    deliberately NOT in :data:`TERMINAL_MISSION_STATUSES`, so subsequent
+    lifecycle events (e.g. a fresh ``MissionCompleted``) are processed
+    normally rather than flagged as post-terminal anomalies.
+    """
 
     ACTIVE = "active"
     COMPLETED = "completed"
     CANCELLED = "cancelled"
+    REOPENED = "reopened"
 
 
 TERMINAL_MISSION_STATUSES: FrozenSet[MissionStatus] = frozenset({
@@ -437,6 +446,64 @@ def _process_mission_event(
                     reason="Invalid MissionClosed payload",
                 )
             )
+        return mission_id, mission_status, mission_type, current_phase
+
+    # Post-mission events (MissionReopened / FollowUpRecorded) are valid ONLY
+    # after the mission has reached a terminal/completed state. They must be
+    # handled BEFORE the generic terminal-state guard below — otherwise that
+    # guard would misfire and flag these by-design post-completion facts as
+    # "Event after terminal state" anomalies.
+    #
+    # The contract is symmetric: a post-mission event that arrives when the
+    # mission is NOT terminal (no prior completion/cancellation) is itself the
+    # anomaly ("post-mission event before completion").
+    if event.event_type == MISSION_REOPENED:
+        try:
+            MissionReopenedPayload(**event.payload)
+        except Exception:
+            anomalies.append(
+                LifecycleAnomaly(
+                    event_id=event.event_id,
+                    event_type=event.event_type,
+                    reason="Invalid MissionReopened payload",
+                )
+            )
+            return mission_id, mission_status, mission_type, current_phase
+        if mission_status not in TERMINAL_MISSION_STATUSES:
+            anomalies.append(
+                LifecycleAnomaly(
+                    event_id=event.event_id,
+                    event_type=event.event_type,
+                    reason="MissionReopened before completion (mission not terminal)",
+                )
+            )
+            return mission_id, mission_status, mission_type, current_phase
+        # Valid re-open: transition out of terminal back to an actionable state.
+        mission_status = MissionStatus.REOPENED
+        return mission_id, mission_status, mission_type, current_phase
+
+    if event.event_type == FOLLOW_UP_RECORDED:
+        try:
+            FollowUpRecordedPayload(**event.payload)
+        except Exception:
+            anomalies.append(
+                LifecycleAnomaly(
+                    event_id=event.event_id,
+                    event_type=event.event_type,
+                    reason="Invalid FollowUpRecorded payload",
+                )
+            )
+            return mission_id, mission_status, mission_type, current_phase
+        if mission_status not in TERMINAL_MISSION_STATUSES:
+            anomalies.append(
+                LifecycleAnomaly(
+                    event_id=event.event_id,
+                    event_type=event.event_type,
+                    reason="FollowUpRecorded before completion (mission not terminal)",
+                )
+            )
+            return mission_id, mission_status, mission_type, current_phase
+        # Valid follow-up: a recorded fact; mission_status is UNCHANGED.
         return mission_id, mission_status, mission_type, current_phase
 
     # Check: event after terminal state

--- a/tests/integration/test_lifecycle_replay.py
+++ b/tests/integration/test_lifecycle_replay.py
@@ -17,17 +17,20 @@ from ulid import ULID
 
 from spec_kitty_events.decisionpoint import reduce_decision_point_events
 from spec_kitty_events.lifecycle import (
+    FOLLOW_UP_RECORDED,
     MISSION_CANCELLED,
     MISSION_COMPLETED,
+    MISSION_REOPENED,
     MISSION_STARTED,
     PHASE_ENTERED,
     REVIEW_ROLLBACK,
+    FollowUpRecordedPayload,
     MissionCancelledPayload,
     MissionCompletedPayload,
+    MissionReopenedPayload,
     MissionStartedPayload,
     MissionStatus,
     PhaseEnteredPayload,
-    ReducedMissionState,
     ReviewRollbackPayload,
     reduce_lifecycle_events,
 )
@@ -409,6 +412,128 @@ class TestFullLifecycleAllEventTypes:
         assert state.mission_status == MissionStatus.COMPLETED
         assert len(state.anomalies) == 1
         assert "terminal" in state.anomalies[0].reason.lower()
+
+    def test_completed_then_reopened_leaves_terminal(self) -> None:
+        """Post-completion MissionReopened is valid and exits terminal."""
+        corr_id = str(ULID())
+        events = [
+            _make_event(
+                MISSION_STARTED,
+                MissionStartedPayload(
+                    mission_id="M001",
+                    mission_type="software-dev",
+                    initial_phase="specify",
+                    actor="user-1",
+                ).model_dump(),
+                lamport_clock=1,
+                corr_id=corr_id,
+            ),
+            _make_event(
+                MISSION_COMPLETED,
+                MissionCompletedPayload(
+                    mission_id="M001",
+                    mission_type="software-dev",
+                    final_phase="review",
+                    actor="user-1",
+                ).model_dump(),
+                lamport_clock=2,
+                corr_id=corr_id,
+            ),
+            _make_event(
+                MISSION_REOPENED,
+                MissionReopenedPayload(
+                    mission_id="M001",
+                    mission_slug="mission-reopen-followup",
+                    reason="Land a follow-up fix",
+                    reopened_by="stijn",
+                    reopened_at="2026-06-14T12:00:00+00:00",
+                ).model_dump(),
+                lamport_clock=3,
+                corr_id=corr_id,
+            ),
+        ]
+        state = reduce_lifecycle_events(events)
+        assert state.mission_status == MissionStatus.REOPENED
+        assert len(state.anomalies) == 0
+        # Replay determinism.
+        assert reduce_lifecycle_events(list(events)) == state
+
+    def test_completed_then_follow_up_keeps_status(self) -> None:
+        """Post-completion FollowUpRecorded is valid; status unchanged."""
+        corr_id = str(ULID())
+        events = [
+            _make_event(
+                MISSION_STARTED,
+                MissionStartedPayload(
+                    mission_id="M001",
+                    mission_type="software-dev",
+                    initial_phase="specify",
+                    actor="user-1",
+                ).model_dump(),
+                lamport_clock=1,
+                corr_id=corr_id,
+            ),
+            _make_event(
+                MISSION_COMPLETED,
+                MissionCompletedPayload(
+                    mission_id="M001",
+                    mission_type="software-dev",
+                    final_phase="review",
+                    actor="user-1",
+                ).model_dump(),
+                lamport_clock=2,
+                corr_id=corr_id,
+            ),
+            _make_event(
+                FOLLOW_UP_RECORDED,
+                FollowUpRecordedPayload(
+                    mission_id="M001",
+                    mission_slug="mission-reopen-followup",
+                    follow_up_type="pr",
+                    pr_number=1926,
+                    recorded_by="stijn",
+                    recorded_at="2026-06-14T12:00:00+00:00",
+                ).model_dump(),
+                lamport_clock=3,
+                corr_id=corr_id,
+            ),
+        ]
+        state = reduce_lifecycle_events(events)
+        assert state.mission_status == MissionStatus.COMPLETED
+        assert len(state.anomalies) == 0
+
+    def test_reopen_before_completion_is_anomaly(self) -> None:
+        """Pre-completion MissionReopened is the anomaly (inverse contract)."""
+        corr_id = str(ULID())
+        events = [
+            _make_event(
+                MISSION_STARTED,
+                MissionStartedPayload(
+                    mission_id="M001",
+                    mission_type="software-dev",
+                    initial_phase="specify",
+                    actor="user-1",
+                ).model_dump(),
+                lamport_clock=1,
+                corr_id=corr_id,
+            ),
+            _make_event(
+                MISSION_REOPENED,
+                MissionReopenedPayload(
+                    mission_id="M001",
+                    mission_slug="mission-reopen-followup",
+                    reason="Premature re-open",
+                    reopened_by="stijn",
+                    reopened_at="2026-06-14T12:00:00+00:00",
+                ).model_dump(),
+                lamport_clock=2,
+                corr_id=corr_id,
+            ),
+        ]
+        state = reduce_lifecycle_events(events)
+        assert state.mission_status == MissionStatus.ACTIVE
+        assert len(state.anomalies) == 1
+        assert "before completion" in state.anomalies[0].reason
 
 
 # ── V1 DecisionPoint golden replay tests ─────────────────────────────────────

--- a/tests/unit/test_lifecycle.py
+++ b/tests/unit/test_lifecycle.py
@@ -1037,6 +1037,163 @@ class TestReduceLifecycleEvents:
         assert "Duplicate" in result.anomalies[0].reason
 
 
+# ── Post-mission reducer semantics (MissionReopened / FollowUpRecorded) ───────
+
+
+class TestPostMissionReducerSemantics:
+    """reduce_lifecycle_events() post-completion semantics.
+
+    Contract: MissionReopened and FollowUpRecorded are valid ONLY after the
+    mission reached a terminal state. Post-completion → no anomaly; a re-open
+    leaves terminal (REOPENED), a follow-up is a recorded fact (status
+    unchanged). Pre-completion → the event itself is the anomaly.
+    """
+
+    def _started(self, lamport: int = 1) -> Event:
+        return _make_mission_event(
+            MISSION_STARTED,
+            MissionStartedPayload(
+                mission_id="M001",
+                mission_type="software-dev",
+                initial_phase="specify",
+                actor="user-1",
+            ).model_dump(),
+            lamport_clock=lamport,
+        )
+
+    def _completed(self, lamport: int = 2) -> Event:
+        return _make_mission_event(
+            MISSION_COMPLETED,
+            MissionCompletedPayload(
+                mission_id="M001",
+                mission_type="software-dev",
+                final_phase="review",
+                actor="user-1",
+            ).model_dump(),
+            lamport_clock=lamport,
+        )
+
+    def _reopened(self, lamport: int = 3) -> Event:
+        return _make_mission_event(
+            MISSION_REOPENED,
+            MissionReopenedPayload(
+                mission_id="M001",
+                mission_slug="mission-reopen-followup",
+                reason="Operator requested re-open to land follow-up fix",
+                reopened_by="stijn",
+                reopened_at="2026-06-14T12:00:00+00:00",
+            ).model_dump(),
+            lamport_clock=lamport,
+        )
+
+    def _follow_up(self, lamport: int = 3) -> Event:
+        return _make_mission_event(
+            FOLLOW_UP_RECORDED,
+            FollowUpRecordedPayload(
+                mission_id="M001",
+                mission_slug="mission-reopen-followup",
+                follow_up_type="commit",
+                commit_sha="a" * 40,
+                recorded_by="stijn",
+                recorded_at="2026-06-14T12:00:00+00:00",
+            ).model_dump(),
+            lamport_clock=lamport,
+        )
+
+    def test_reopen_after_completion_is_not_anomaly(self) -> None:
+        result = reduce_lifecycle_events(
+            [self._started(), self._completed(), self._reopened()]
+        )
+        assert result.anomalies == ()
+        # Re-open transitions the mission OUT of terminal.
+        assert result.mission_status == MissionStatus.REOPENED
+        assert result.mission_status not in TERMINAL_MISSION_STATUSES
+        assert result.event_count == 3
+
+    def test_follow_up_after_completion_is_not_anomaly_and_status_unchanged(
+        self,
+    ) -> None:
+        result = reduce_lifecycle_events(
+            [self._started(), self._completed(), self._follow_up()]
+        )
+        assert result.anomalies == ()
+        # FollowUpRecorded is a recorded fact: status stays terminal.
+        assert result.mission_status == MissionStatus.COMPLETED
+        assert result.event_count == 3
+
+    def test_reopen_before_completion_is_anomaly(self) -> None:
+        result = reduce_lifecycle_events([self._started(), self._reopened(lamport=2)])
+        assert result.mission_status == MissionStatus.ACTIVE
+        assert len(result.anomalies) == 1
+        assert result.anomalies[0].event_type == MISSION_REOPENED
+        assert "before completion" in result.anomalies[0].reason
+
+    def test_follow_up_before_completion_is_anomaly(self) -> None:
+        result = reduce_lifecycle_events(
+            [self._started(), self._follow_up(lamport=2)]
+        )
+        assert result.mission_status == MissionStatus.ACTIVE
+        assert len(result.anomalies) == 1
+        assert result.anomalies[0].event_type == FOLLOW_UP_RECORDED
+        assert "before completion" in result.anomalies[0].reason
+
+    def test_reopen_then_complete_again_processes_normally(self) -> None:
+        """After a valid re-open, status is non-terminal so a fresh
+        MissionCompleted is applied without a post-terminal anomaly."""
+        result = reduce_lifecycle_events(
+            [
+                self._started(),
+                self._completed(lamport=2),
+                self._reopened(lamport=3),
+                self._completed(lamport=4),
+            ]
+        )
+        assert result.anomalies == ()
+        assert result.mission_status == MissionStatus.COMPLETED
+        assert result.event_count == 4
+
+    def test_invalid_reopen_payload_after_completion_is_anomaly(self) -> None:
+        bad = _make_mission_event(
+            MISSION_REOPENED,
+            {"mission_id": "M001"},  # missing required fields
+            lamport_clock=3,
+        )
+        result = reduce_lifecycle_events([self._started(), self._completed(), bad])
+        assert result.mission_status == MissionStatus.COMPLETED
+        assert len(result.anomalies) == 1
+        assert "Invalid MissionReopened payload" == result.anomalies[0].reason
+
+    def test_invalid_follow_up_payload_after_completion_is_anomaly(self) -> None:
+        bad = _make_mission_event(
+            FOLLOW_UP_RECORDED,
+            {"mission_id": "M001", "follow_up_type": "commit"},  # missing commit_sha
+            lamport_clock=3,
+        )
+        result = reduce_lifecycle_events([self._started(), self._completed(), bad])
+        assert result.mission_status == MissionStatus.COMPLETED
+        assert len(result.anomalies) == 1
+        assert "Invalid FollowUpRecorded payload" == result.anomalies[0].reason
+
+    def test_follow_up_does_not_leave_terminal_for_next_guard(self) -> None:
+        """A follow-up keeps the mission terminal, so a later by-design
+        anomaly event (e.g. PhaseEntered) is still flagged."""
+        late_phase = _make_mission_event(
+            PHASE_ENTERED,
+            PhaseEnteredPayload(
+                mission_id="M001",
+                phase_name="implement",
+                actor="user-1",
+            ).model_dump(),
+            lamport_clock=4,
+        )
+        result = reduce_lifecycle_events(
+            [self._started(), self._completed(), self._follow_up(), late_phase]
+        )
+        assert result.mission_status == MissionStatus.COMPLETED
+        assert len(result.anomalies) == 1
+        assert "terminal" in result.anomalies[0].reason.lower()
+
+
 # ── MissionReopenedPayload ───────────────────────────────────────────────────
 
 

